web/src/utils/useInterval.tsx

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright © 2020 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { useEffect, useRef } from 'react';
+
+export default function useInterval(callback: any, delay: number) {
+    const savedCallback = useRef();
+
+    useEffect(() => {
+        savedCallback.current = callback;
+    }, [callback]);
+
+    useEffect(() => {
+        function tick() {
+            //@ts-ignore
+            savedCallback.current();
+        }
+        if (delay !== null) {
```

Comment:
I tend to prefer early returns in this case; not sure if its better or just different but I **think** early returns are generally preferred.
```
if (delay === null) { 
    return;
}
```